sdk/core/Azure.Core/src/JsonData.cs

```diff
@@ -0,0 +1,806 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Dynamic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure
+{
+    /// <summary>
+    /// A mutable representation of a JSON value.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [DebuggerTypeProxy(typeof(JsonDataDebuggerProxy))]
+    public class JsonData : IDynamicMetaObjectProvider, IEquatable<JsonData>
+    {
+        private readonly JsonValueKind _kind;
+
+        private object? _value;
+        private Dictionary<string, JsonData>? _objectRepresentation;
+        private List<JsonData>? _arrayRepresentation;
+
+        private static readonly JsonSerializerOptions DefaultJsonSerializerOptions = new JsonSerializerOptions();
+
+        /// <summary>
+        /// Parses a UTF8 encoded string representing a single JSON value into a <see cref="JsonData"/>.
+        /// </summary>
+        /// <param name="utf8Json">A UTF8 encoded string representing a JSON value.</param>
+        /// <returns>A <see cref="JsonData"/> representation of the value.</returns>
+        internal static JsonData Parse(BinaryData utf8Json) => new JsonData(JsonDocument.Parse(utf8Json));
```

Comment:
Handle IDisposable properly.

Reply:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L61

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L87

Reply:
Per @AlexanderSher:

> I've created a small benchmark. It looks like disposing JsonDocument is quite important:
> https://gist.github.com/AlexanderSher/251a035cfbbe539337e4a1c6504ff52b

Reply:
@AlexanderSher, does this address your concern on this? https://github.com/Azure/azure-sdk-for-net/pull/31769/commits/795e2ff92cc19240f588d6595594e5b661ea2c08